### ISSUE TITLE
v4: custom <Admin /> removes all labels (i18n?)

### ISSUE THREAD

**Issue by RWOverdijk** [marmelab/react-admin#7520]
When using the AdminContext instead of `<Admin />` all labels change into what I assume are keys.

Upon checking the code [I think this is the issue](https://github.com/marmelab/react-admin/blob/v4.0.0-rc.1/packages/react-admin/src/Admin.tsx#L146,L149):

```js
Admin.defaultProps = {
    i18nProvider: defaultI18nProvider,
    store: localStorageStore(),
};
```

These aren't mentioned in the docs. I think this should be added to the docs, or should be moved into `<AdminContext />`.

**What you were expecting:**

I was expecting nothing to change and everything to look the same.

**What happened instead:**

After following this guide: https://marmelab.com/react-admin/doc/4.0/Admin.html#unplugging-the-admin-using-admincontext-and-adminui

I got the following result:

![image](https://user-images.githubusercontent.com/781745/162771388-0933902d-764f-4671-bff2-6f174dda562a.png)

**Related code:**

My <App /> looks like this now (removed most resources and the irrelevant bits for brevity):

```tsx
function App() {
  return (
    <AdminContext authProvider={authProvider} dataProvider={withUpload}>
      <AsyncResources />
    </AdminContext>
  );
}

function AsyncResources() {
  return (
    <AdminUI layout={MyLayout} loginPage={SignIn} disableTelemetry={true}>
      <Resource name="profiles" list={ProfileList} show={ProfileShow} edit={ProfileEdit} create={ProfileCreate} />
      <Resource name="users" list={UserList} show={UserShow} edit={UserEdit} create={UserCreate} />
    </AdminUI>
  );
}
```

**Environment**

* React-admin version: 4.0.0-rc.1
* React version: 17.0.2
* Browser: Chrome 
* Stack trace (in case of a JS error): N/A

**Comment by fzaninotto** [comment]
You're right. I'm marking this as a documentation bug. 

Would you like to submit a PR to fix it?

**Comment by RWOverdijk** [comment]
@fzaninotto https://github.com/marmelab/react-admin/pull/7524

You know I've been around for a while because of the fork name lol

**Comment by djhi** [comment]
Wouldn't it be better to set those defaults directly on the `AdminContext`?

**Comment by fzaninotto** [comment]
> Wouldn't it be better to set those defaults directly on the `AdminContext`?

No, because it would create a dependency between `ra-material-ui` and `ra-i18n-polyglot`

**Comment by fzaninotto** [comment]
Fixed by #7524
